SignalMessaging/attachments/ApproveContactShareViewController.swift

```diff
@@ -392,11 +436,26 @@ public class ApproveContactShareViewController: OWSViewController, EditContactSh
     // MARK: -
 
     func filteredContactShare() -> ContactShareViewModel {
+        var avatarImage: UIImage?
```

Comment:
There's definitely some friction around the ViewModel vs. Model in terms of `UIImage` vs. `NSData` and updates.  It'd also be nice to avoid unnecessary image conversions, but we're not doing that - yet.  e.g. if system contact is a jpg, we're reconverting it to jpg when we send the contact share which isn't ideal.

Reply:
Right - I wonder if makes sense to move any responsibility that must maintain a reference to the UIImage to the ViewModel, then hopefully we could avoid creating these one-off db connections and creating write transactions on the main thread.

Reply:
> Right - I wonder if makes sense to move any responsibility that must maintain a reference to the UIImage to the ViewModel, then hopefully we could avoid creating these one-off db connections and creating write transactions on the main thread.

The VM owns the UIImage and I believe we are no longer creating any sync r/w transactions on the main thread.